### DISCUSSION TITLE
Is it possible to get the text value of "permalink" instead of an html?

### DISCUSSION THREAD

**Discussion by jaeyson** [slimndap/wp-theatre#334] [Q&A] [answered]
I'm basing on these docs: https://github.com/slimndap/wp-theatre/wiki/Shortcodes#template

Question: Can I extract the value of `{{permalink}}` as text instead of html?

**Comment by slimndap** [comment]
You can try `{{tickets_url}}`. However, the output will still be enclosed in `<div>` tags.

If you need more control, then you need to look at the `wpt/event/template/field/value` filter:
https://github.com/slimndap/wp-theatre/blob/master/functions/wpt_event_template.php#L115

**Reply by jaeyson** [reply]
Thank you for helping me out @slimndap! sorry for bothering you again and just one last question: I still don't get it on how to use that filter https://github.com/slimndap/wp-theatre/blob/e923b0d767d2660d00801a88bcc4fc4732effec2/functions/wpt_event_template.php#L115

does that mean I use it like `{{permalink('html', false)}}`? I just wanted to return it e.g. `https://example.com/id/123`

**Reply by slimndap** [reply, marked as answer]
I doesn't.

This is a WordPress filter and you need to add PHP code to the functions.php of your theme. See: https://developer.wordpress.org/plugins/hooks/filters/

Now that I had another look, it is better to use this filter:
https://github.com/slimndap/wp-theatre/blob/e923b0d767d2660d00801a88bcc4fc4732effec2/functions/wpt_event_template.php#L115

For example, you can add support for a new placeholder, eg `{{raw_tickets_url}}`.

```php
add_filter( 'wpt_event_raw_tickets_url_html', 'my_theme_add_raw_tickets_url_placeholder', 10, 3 );

function my_theme_add_raw_tickets_url_placeholder( $html, $field, $event ) {

  if ( 'raw_tickets_url' != $field ) {
    return $value;
  }

  return get_post_meta( $event→ID, 'tickets_url', true );

}
```

**Reply by jaeyson** [reply]
Wow this solves my problem! Thank you very much @slimndap! and the docs you pointed out makes sense now 🙏🏾🙏🏾🙏🏾
